src/audio/crossover/crossover.c

```diff
@@ -140,7 +140,9 @@ static int crossover_assign_sinks(struct comp_dev *dev,
 		sink_c = buffer_acquire(sink);
 
 		pipeline_id = sink_c->pipeline_id;
-		state = sink_c->sink->state;
+		state = -1;
+		if (sink_c->sink)
+			state = sink_c->sink->state;
 		buffer_release(sink_c);
```

Comment:
'ditto' doesn't help me here, this is a different piece of code @lyakh. What is wrong with my suggested solution?

Reply:
@lyakh ping

Reply:
@lgirdwood I explained this in https://github.com/thesofproject/sof/pull/6131#discussion_r972664350 above

Reply:
Here I'm even less sure than in the mixin-mixout case, but I'd say - yes, also here a sink buffer with a NULL component link should be ignored / skipped. @plbossart have you actually observed any of these triggering? I think these could only potentially be hit in race cases when pipelines get torn down, and in fact maybe we should fix those races instead of plumbing for them. The commit message should be fixed too.

Reply:
no, I have not observed issues personally @lyakh, this correct is just based on a grep search and trying to use a consistent way of accessing sink_c->sink members/fieds.

This component is one of the few were there was a direct dereference. I don't have any idea what this component does, just that it doesn't follow the code pattern others do.